When trying to compile, I get a String error

I'm on Linux, under Swift 4.1

This is what I ran:

```console
felix@felix-X550LD ~/D/P/A/beak> swift build
Compile Swift Module 'BeakCore' (15 sources)
/home/felix/Documents/Programming/AwesomeSystemSwift/beak/Sources/BeakCore/SourceKit.swift:86:54: error: value of type 'String' has no member 'substringWithByteRange'
        let substring = range(for: source).flatMap { contents.substringWithByteRange(start: Int($0.offset), length: Int($0.length)) }
                                                     ^~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~
error: terminated(1): /home/felix/Documents/Programming/Binaries/swift/swift-4.1-snapshot/usr/bin/swift-build-tool -f /home/felix/Documents/Programming/AwesomeSystemSwift/beak/.build/debug.yaml main output:
    

```

[This line](https://github.com/yonaskolb/Beak/blob/f8431845a9844a7a736a5e369025420a5edf0aef/Sources/BeakCore/SourceKit.swift#L86)'s the one causing the error. Maybe the library hasn't been updated to Swift 4.1?

Anyway. Thanks for your help, and I hope you have a good day :)

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,9 +1,22 @@
 version: 2
 jobs:
-  build:
+  macOS_swift_4.1:
     macos:
-      xcode: "9.2.0"
+      xcode: "9.3.0"
     steps:
       - checkout
       - run: swift build
       - run: swift test
+  linux_swift_4.1:
+    docker:
+      - image: norionomura/swift:41
+    steps:
+      - checkout
+      - run: swift build
+      - run: swift test
+workflows:
+  version: 2
+  workflow:
+    jobs:
+      - macOS_swift_4.1
+      - linux_swift_4.1

--- a/README.md
+++ b/README.md
@@ -1,6 +1,7 @@
 # Beak 🐦
 
 [![SPM](https://img.shields.io/badge/Swift_PM-compatible-brightgreen.svg?style=for-the-badge)](https://swift.org/package-manager)
+![Linux](https://img.shields.io/badge/Platforms-macOS_Linux-blue.svg?style=for-the-badge)
 [![Git Version](https://img.shields.io/github/release/yonaskolb/Beak.svg?style=for-the-badge)](https://github.com/yonaskolb/Beak/releases)
 [![Build Status](https://img.shields.io/circleci/project/github/yonaskolb/Beak.svg?style=for-the-badge)](https://circleci.com/gh/yonaskolb/Beak)
 [![license](https://img.shields.io/github/license/yonaskolb/Beak.svg?style=for-the-badge)](https://github.com/yonaskolb/Beak/blob/master/LICENSE)

--- a/Sources/BeakCore/SourceKit.swift
+++ b/Sources/BeakCore/SourceKit.swift
@@ -83,7 +83,7 @@ enum Substring {
     }
 
     func extract(from source: [String: SourceKitRepresentable], contents: String) -> String? {
-        let substring = range(for: source).flatMap { contents.substringWithByteRange(start: Int($0.offset), length: Int($0.length)) }
+        let substring = range(for: source).flatMap { contents.substring(start: Int($0.offset), length: Int($0.length)) }
         return substring?.isEmpty == true ? nil : substring?.trimmingCharacters(in: .whitespacesAndNewlines)
     }
 }

--- a/Tests/BeakTests/XCTestManifests.swift
+++ b/Tests/BeakTests/XCTestManifests.swift
@@ -1,0 +1,15 @@
+import XCTest
+
+extension BeakTests {
+    static let __allTests = [
+        ("testBeak", testBeak),
+    ]
+}
+
+#if !os(macOS)
+public func __allTests() -> [XCTestCaseEntry] {
+    return [
+        testCase(BeakTests.__allTests),
+    ]
+}
+#endif

--- a/Tests/LinuxMain.swift
+++ b/Tests/LinuxMain.swift
@@ -1,0 +1,8 @@
+import XCTest
+
+import BeakTests
+
+var tests = [XCTestCaseEntry]()
+tests += BeakTests.__allTests()
+
+XCTMain(tests)

